deployment/windows/jellyfin.nsi

```diff
@@ -0,0 +1,202 @@
+    !insertmacro MUI_PAGE_DIRECTORY
+	
+    !insertmacro MUI_PAGE_INSTFILES
+
+    !insertmacro MUI_UNPAGE_CONFIRM
+    !insertmacro MUI_UNPAGE_INSTFILES
+  
+;--------------------------------
+;Languages
+ 
+    !insertmacro MUI_LANGUAGE "English"
+
+;--------------------------------
+;Installer Sections
+
+Section "Install Jellyfin (required)" InstallJellyfin
+    SetOutPath "$INSTDIR"
+    SetShellVarContext current
+    StrCpy $_JELLYFINDATADIR_ "$LOCALAPPDATA\jellyfin\"
+
+; Pack all the files that were just compiled
+    File /r $%InstallLocation%\* 
+	
+; Write the installation path into the registry
+    WriteRegStr HKLM "Software\Jellyfin" "InstallLocation" "$INSTDIR"
+
+; Write the uninstall keys for Windows
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "DisplayName" "Jellyfin $_JELLYFINVERSION_"
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "UninstallString" '"$INSTDIR\Uninstall.exe"'
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "DisplayIcon" '"$INSTDIR\Jellyfin.exe",0'
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "Publisher" "The Jellyfin project"
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "URLInfoAbout" "https://jellyfin.github.io/"
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "DisplayVersion" "$_JELLYFINVERSION_"
+    WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "NoModify" 1
+    WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\Jellyfin" "NoRepair" 1
+
+;Create uninstaller
+    WriteUninstaller "$INSTDIR\Uninstall.exe"
+
+SectionEnd
+
+Section "Jellyfin desktop shortcut" DesktopShortcut
+    SetShellVarContext current
+    DetailPrint "Creating desktop shortcut"
+    CreateShortCut "$DESKTOP\Jellyfin.lnk" "$INSTDIR\jellyfin.exe"
+SectionEnd
+
+Section /o "Jellyfin Service" InstallService
+    ExecWait '"$INSTDIR\nssm.exe" install Jellyfin "$INSTDIR\jellyfin.exe" --datadir "$_JELLYFINDATADIR_"' $0
+    DetailPrint "Jellyfin Service install, $0"
+    Sleep 3000
+    ExecWait '"$INSTDIR\nssm.exe" set Jellyfin Start SERVICE_DELAYED_AUTO_START' $0
+    DetailPrint "Jellyfin Service setting, $0"
+    StrCpy $_SERVICEINSTALLED_ "YES"
+SectionEnd
+
+Section /o "Start Jellyfin after installation" LaunchJellyfin
+; either start the service or launch jellyfin standalone
+    StrCmp $_SERVICEINSTALLED_ "YES" ServiceStart Standalone
+    
+    ServiceStart:
+    ExecWait '"$INSTDIR\nssm.exe" start Jellyfin' $0
+    DetailPrint "Jellyfin service start, $0"
+    Return
+    
+    Standalone:
+    ExecWait '"$INSTDIR"\jellyfin.exe' $0
+    DetailPrint "$INSTDIR\jellyfin.exe start, $0"
+
+SectionEnd
+
+;TODO
+; This section hasn't been tested completely
+Section /o "Migrate Emby Library" MigrateEmbyLibrary
```

Comment:
We probably don't want this - see above.

Reply:
Agree